Makefile

```
@@ -368,6 +371,53 @@ helm-chart: manifests kustomize helmify
 unit-test: ## Run unit tests for the helm chart
 	PATH="$(CURDIR)/scripts:$(PATH)" bats test/unit/
 
+##@ EKS
+.PHONY: create-eks
+create-eks: ## Create a new EKS cluster
+	$(TERRAFORM) -chdir=$(TF_EKS_DIR) init -upgrade
+	$(TERRAFORM) -chdir=$(TF_EKS_DIR) apply -auto-approve
+	$(AWS) eks --region $$($(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw region) update-kubeconfig \
+    --name $$($(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw cluster_name)
+
+.PHONY: ci-ecr-build
+ci-ecr-build: ## Build operator binary (without generating assets).
+	CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) go build \
+		-ldflags "${LD_FLAGS} $(shell GOOS=$(GOOS) GOARCH=$(GOARCH) ./scripts/ldflags-version.sh)" \
+		-a \
+		-o $(BUILD_DIR)/$(GOOS)/$(GOARCH)/$(BIN_NAME) \
+		.
+
+# Currently only supports amd64
+.PHONY: ci-ecr-build-push
+ci-ecr-build-push: ## Build the operator image and push it to the ECR repository
+	@$(eval IMG := $(shell $(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw ecr_url):$(VERSION))
+	$(MAKE) ci-ecr-build GOOS=$(GOOS) GOARCH=$(GOARCH)
+	docker build -t $(IMG) . --target release-default --build-arg GO_VERSION=$(shell cat .go-version) --build-arg TARGETOS=$(GOOS) --build-arg TARGETARCH=$(GOARCH)
+	$(AWS) ecr get-login-password --region $$($(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw region) | docker login --username AWS --password-stdin $$($(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw ecr_url)
+	docker push $(IMG)
+
+.PHONY: integration-test-eks
+integration-test-eks: ## Run unit tests in the EKS cluster
+	@$(eval KIND_CLUSTER_CONTEXT := $(shell $(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw cluster_arn))
+	@$(eval IMAGE_TAG_BASE := $(shell $(TERRAFORM) -chdir=$(TF_EKS_DIR) output -raw ecr_url))
+	$(MAKE) setup-vault KIND_CLUSTER_CONTEXT=$(KIND_CLUSTER_CONTEXT) IMAGE_TAG_BASE=$(IMAGE_TAG_BASE)
+	kubectl port-forward vault-0 -n $(K8S_VAULT_NAMESPACE) 38300:8200 &
+	cd $(CONFIG_MANAGER_DIR) && $(KUSTOMIZE) edit set image controller=$(IMG)
+	SUPPRESS_TF_OUTPUT=$(SUPPRESS_TF_OUTPUT) SKIP_CLEANUP=$(SKIP_CLEANUP) OPERATOR_NAMESPACE=$(OPERATOR_NAMESPACE) \
+	OPERATOR_IMAGE_REPO=$(IMAGE_TAG_BASE) OPERATOR_IMAGE_TAG=$(VERSION) \
+    INTEGRATION_TESTS=true KIND_CLUSTER_NAME=$(KIND_CLUSTER_NAME) KIND_CLUSTER_CONTEXT=$(KIND_CLUSTER_CONTEXT) CGO_ENABLED=0 \
+	go test github.com/hashicorp/vault-secrets-operator/test/integration/... $(TESTARGS) -timeout=30m
```

I wonder if we could just call `make integration-test` here with the cluster context. Something like this?

```suggestion
	$(MAKE) integration-test KIND_CLUSTER_CONTEXT=$(KIND_CLUSTER_CONTEXT)
```

Though I'm also thinking we may want to test Vault Enterprise on EKS instead of OSS.

---

@tvoran - Should it be either OSS or Ent? or should I look at doing both tests perhaps? I'll try out this recommendation, I believe I'll also need to pass the Operator Image repo and Image tag in that make command

---

Doing both would be fine too, but I think we should at least be testing Ent.

---

Add the enterprise tests + OSS tests in the CI jobs using env var and the same make command, working fine 👍🏾  I'll add it in the readme file

Fixed the port-forward, and made it universal so that when I work with GKE and AKS it should work there as well.

now integration-test-eks calls integration-test as it should! 👍🏾